u8timeseries/models/tcn_model.py

```diff
@@ -0,0 +1,254 @@
+"""
+Temporal Convolutional Network
+------------------------------
+"""
+
+import numpy as np
+import math
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from typing import Optional
+
+from ..timeseries import TimeSeries
+from ..logging import raise_if_not, get_logger
+from .torch_forecasting_model import TorchForecastingModel, _TimeSeriesDataset1DShifted
+
+logger = get_logger(__name__)
+
+
+class _ResidualBlock(nn.Module):
+
+    def __init__(self, num_filters, kernel_size, dilation_base, dropout, weight_norm, i, num_layers):
+        """ PyTorch module implementing a residual block module used in `_TCNModule`.
+
+        Parameters
+        ----------
+        num_filters
+            The number of filters in a convolutional layer of the TCN.
+        kernel_size
+            The size of every kernel in a convolutional layer.
+        dilation_base
+            The base of the exponent that will determine the dilation on every level.
+        dropout
+            The dropout rate for every convolutional layer.
+        weight_norm
+            Boolean value indicating whether to use weight normalization.
+        i
```

Comment:
could we find maybe a better name? For instance `nr_blocks_below`

Reply:
done